packages/build-tools/src/utils/outputs.ts

```diff
@@ -0,0 +1,80 @@
+import { Generic } from '@expo/eas-build-job';
+import { BuildStep, jsepEval } from '@expo/steps';
+import { bunyan } from '@expo/logger';
+import nullthrows from 'nullthrows';
+import fetch from 'node-fetch';
+
+import { BuildContext } from '../context';
+
+export async function uploadJobOutputsToWwwAsync(
+  ctx: BuildContext<Generic.Job>,
+  {
+    steps,
+    logger,
+    expoApiV2BaseUrl,
+  }: { steps: BuildStep[]; logger: bunyan; expoApiV2BaseUrl: string }
+): Promise<void> {
+  if (!ctx.job.outputs) {
+    logger.info('Job defines no outputs, skipping upload');
+    return;
+  }
+
+  try {
+    const workflowJobId = nullthrows(ctx.job.builderEnvironment?.env?.__WORKFLOW_JOB_ID);
+    const robotAccessToken = nullthrows(ctx.job.secrets?.robotAccessToken);
+
+    const interpolationContext = {
+      steps: Object.fromEntries(steps.map((step) => [step.id, getDynamicValuesFromStep(step)])),
+    };
+    logger.info({ dynamicValues: interpolationContext }, 'Using dynamic values');
+
+    const outputs = getJobOutputsFromSteps({
+      jobOutputDefinitions: ctx.job.outputs,
+      interpolationContext,
+    });
+    logger.info('Uploading outputs');
+
+    await fetch(new URL(`workflows/${workflowJobId}`, expoApiV2BaseUrl).toString(), {
+      method: 'PATCH',
+      body: JSON.stringify({ outputs }),
+      headers: {
+        Authorization: `Bearer ${robotAccessToken}`,
+        'Content-Type': 'application/json',
+      },
+      timeout: 20000,
+    });
+  } catch (err) {
+    logger.error({ err }, 'Failed to upload outputs');
```

Comment:
Shouldn't we fail the workflow run altogether when an error is thrown? If you expect outputs from this workflow in your other runs I guess it would be hard to debug later on if we mark this run as a `success` but some required inputs will be missing.

What we can do is retry outputs upload a couple of times and if it still fails throw an error and mark workflow as errored.